api/v1alpha1/envoygateway_types.go

```diff
@@ -512,6 +512,14 @@ type ExtensionManager struct {
 	//
 	// +optional
 	FailOpen bool `json:"failOpen,omitempty"`
+
+	// MaxMessageSizeBytes defines the maximum message size in bytes that can be
+	// sent to or received from the Extension Service.
+	// Default: 4 MB
+	// +kubebuilder:validation:Minimum=1
+	//
+	// +optional
+	MaxMessageSizeBytes *int `json:"grpcMaxMessageSize,omitempty"`
```

Comment:
the json tag is different from the field name

also, can you check if we use the `bytes` suffix in other places of the API (e.g. buffer limit) or is it implied and part of the doc string ? would be good to keep it consistent